stint/io.nim

```diff
@@ -226,9 +226,10 @@ func skipPrefixes(current_idx: var int, str: string, radix: range[2..16]) {.inli
         raise newException(ValueError, "Parsing mismatch, 0o prefix is only valid for an octal number (base 8)")
       current_idx = 2
     elif str[1] in {'b', 'B'}:
-      if radix != 2:
-        raise newException(ValueError, "Parsing mismatch, 0b prefix is only valid for a binary number (base 2)")
-      current_idx = 2
+      if radix notin {2, 16}:
```

Comment:
```suggestion
      if radix == 2:
        current_idx = 2
      elif radix != 16:
        raise newException(ValueError, "Parsing mismatch, 0b prefix is only valid for a binary number (base 2) or as first byte of a hexadecimal number (base 16)")
        
```

Reply:
I personally don't find it harder to read since the `if/elif` pattern is closer to a switch statement, so you clearly see how the various `radix` cases are handled.

Reply:
OK, looks great. Now, we only need new test cases that will cover the problematic case.